src/main/java/net/snowflake/client/jdbc/SnowflakeUtil.java

```diff
@@ -395,7 +395,13 @@ static List<FieldMetadata> createFieldsMetadata(
       throws SnowflakeSQLLoggedException {
     List<FieldMetadata> fields = new ArrayList<>();
     for (JsonNode node : fieldsJson) {
-      String colName = node.path("name").asText();
+      String colName;
+      if (!node.path("fieldType").isEmpty()) {
+        colName = node.path("fieldName").asText();
+        node = node.path("fieldType");
```

Comment:
Can you explain it a bit? I don't understand this situation, when should enter `if` and `else` branches.